conda/core/index.py

```diff
@@ -371,64 +465,61 @@ def fetch_repodata(url, cache_dir=None, use_cache=False, session=None):
         if (timeout > 0 or context.offline) and not url.startswith('file://'):
             log.debug("Using cached repodata for %s at %s. Timeout in %d sec",
                       url, cache_path, timeout)
-            return read_local_repodata(cache_path)
+            return read_local_repodata(cache_path, url, schannel, priority)
 
         log.debug("Locally invalidating cached repodata for %s at %s", url, cache_path)
 
     try:
         assert url is not None, url
-        fetched_repodata = fetch_repodata_remote_request(session, url,
-                                                         mod_etag_headers.get('_etag'),
-                                                         mod_etag_headers.get('_mod'))
+        repodata = fetch_repodata_remote_request(session, url,
+                                                 mod_etag_headers.get('_etag'),
+                                                 mod_etag_headers.get('_mod'))
     except Response304ContentUnchanged:
```

Comment:
We need a way of returning a `304 NOT MODIFIED` equivalent for file-based urls.  Or more accurately, because of the logic here, the pickling strategy is bypassing all file-based urls.  We need a way to know if we can use the pickled repodata.